tests/conftest.py

```diff
@@ -0,0 +1,33 @@
+import pytest
+import psycopg2
+
+
+@pytest.fixture
+def db_connection():
+    connection = psycopg2.connect(
+        host="localhost",
+        database="edison",
+        user="postgres",
+        password="edison")
+    
+    yield connection
+    connection.close()
+
+@pytest.fixture
+def cursor(db_connection):
+    cursor = db_connection.cursor()
+    yield cursor
+    db_connection.rollback()
+
+@pytest.fixture
+def build_users_table(cursor):
+    cursor.execute("""
+    INSERT INTO users (username) VALUES (%s);
+    """,
+    ("ahinoam",))
+
+@pytest.fixture
+def reset_users_primary_key(cursor):
```

Comment:
Hey @ifireball 😄 
I wrote few tests for checking the DB like you asked me in the last PR.
Can you please help me to find a way to reset the primary key sequence after the tests will be finished?
I tried to enter this fixture in the 'cursor' fixture (right after the 'yield'), but it didn't work, I assume it happened because after the 'yield' the cursor is not available anymore?

Reply:
One last comment. The tests that you wrote here verify basic database functionality, which doesn't need to be tested.
What you should probably test is:
* the database exists
* you can connect to it
* the tables you created exist.

Any other tests are really redundant.

Reply:
1. I want to reset the primary key in order to clean up my database after the tests and return it to the initial state. I think it will make more sense when the database show you data that begins with id 1.
2. About the commit after the execute, I used the “cursor.execute” without commit after it but still the query was executed. So why in this case I should add the commit? 
Furthermore, I don’t know where to call this fixture, because if I inject it to “cursor” it will activate it in the beginning of the “cursor” fixture, no?
3. Ack

Reply:
1 - the alter sequence statement is correct and it should work. Try to remember, in the lifetime of an app the database sequence will very rarely stay sequential.
2 - Have you tried executing this code and then going into the database using psql and seeing if the data you added was still there? As far as i know pyscopg does not have automated commits. One possibly confusing thing about sequences and transactions is when you get a new sequence that ignores the transaction, so if you insert and then rollback the transaction, the sequence is still at the next number.
On line 20 you have `db_connection.rollback()` when is this code called? A rollback means erase everything that I've done during this current transaction.

Reply:
> Have you tried executing this code and then going into the database using psql and seeing if the data you added was still there?

I thought that `cursor.execute` is enough because when I ran the test `def test_username(cursor, build_users_table)` it was passed, so the row 'ahinoam' did enter to the DB.
But when I checked it in the DB itself with psql, it didn't appear there (and it's not because `db_connection.rollback()` because I put it in a comment).
Do you have any idea what is the reason for that?

Reply:
> On line 20 you have `db_connection.rollback()` when is this code called? A rollback means erase everything that I've done during this current transaction.

I want it to be called after the cursor will finish his job every test, therefore I put  `db_connection.rollback()` right after the `yield cursor` statement. But I have to be honest I don't understand the `yield` statements into their depth...

Reply:
The way the database works is that during the same transaction everything looks like it is saved, but it really isn't until you call commit. 
If you don't manually call commit, it is the same as calling rollback.